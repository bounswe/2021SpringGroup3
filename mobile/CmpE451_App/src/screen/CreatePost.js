import React, {useEffect, useState} from 'react';
import {
  View,
  Text,
  TouchableOpacity,
  TextInput,
  ScrollView,
  Alert,
} from 'react-native';
import {BackHandler, Platform} from 'react-native';
import {useNavigation} from '@react-navigation/native';
import Header from '../component/Header';
import Icon from 'react-native-vector-icons/Ionicons';
import Icon2 from 'react-native-vector-icons/MaterialIcons';
import {size} from 'lodash';
import {bold} from 'chalk';
import * as Requests from '../util/Reguests';
import {TEXT, PAGE_VARIABLES, BASE_URL} from '../constants';

export default function CreatePost({route}) {
  const {communityName, communityId} = route.params;
  const [dateFieldNames, setDateFieldNames] = useState({name: '', value: ''});
  const [linkFieldNames, setLinkFieldNames] = useState({name: '', value: ''});
  const [locationFieldNames, setLocationFieldNames] = useState({
    name: '',
    value: '',
  });
  const [textFieldNames, setTextFieldNames] = useState({name: '', value: ''});
  const [numberFieldNames, setNumberFieldNames] = useState({
    name: '',
    value: '',
  });
  const navigation = useNavigation();
  function handleBackButtonClick() {
    navigation.goBack();
    return true;
  }
  useEffect(() => {
    if (Platform.OS === 'android') {
      BackHandler.addEventListener('Home', handleBackButtonClick);
      return () => {
        BackHandler.removeEventListener('Home', handleBackButtonClick);
      };
    }
  }, []);

  const cleanEmptyValuesObject = obj => {
    for (var propName in obj) {
      if (
        obj[propName] === '' ||
        obj[propName] === undefined ||
        obj[propName] === null
      ) {
        delete obj[propName];
      }
    }
    return obj;
  };

  useEffect(async () => {
    const postTypeDetail = await Requests.getPostTypeDetail({
      communityId: PAGE_VARIABLES.communityId,
      postTypeId: PAGE_VARIABLES.postTypeId,
    });
    const {
      dateFieldNames,
      id,
      linkFieldNames,
      numberFieldNames,
      locationFieldNames,
      name,
      textFieldNames,
    } = JSON.parse(postTypeDetail);

    console.log('JSON.parse(postTypeDetail);: ', JSON.parse(postTypeDetail));
    if (dateFieldNames != null && dateFieldNames.length > 0) {
      setDateFieldNames({name: dateFieldNames[0], value: ''});
    }
    if (linkFieldNames != null && linkFieldNames.length > 0) {
      setLinkFieldNames({name: linkFieldNames[0], value: ''});
    }

    if (locationFieldNames != null && locationFieldNames.length > 0) {
      setLocationFieldNames({name: locationFieldNames[0], value: ''});
    }

    if (textFieldNames != null && textFieldNames.length > 0) {
      setTextFieldNames({name: textFieldNames[0], value: ''});
    }
    if (numberFieldNames != null && numberFieldNames.length > 0) {
      setNumberFieldNames({name: numberFieldNames[0], value: ''});
    }
  }, []);

  async function createPostHandler() {
    let response = await Requests.createPost(
      cleanEmptyValuesObject({
        postTypeId: PAGE_VARIABLES.postTypeId,
        communityId: PAGE_VARIABLES.communityId,
        textFields: textFieldNames.value === '' ? null : [textFieldNames],
        dateFields: [
          {
            name: 'Date',
            value: '2021-11-16T13:48:40.868+03:00',
          },
        ],
        numberFields: numberFieldNames.value === '' ? null : [numberFieldNames],
        locationFields:
          locationFieldNames.value === '' ? null : [locationFieldNames],
        linkFields: linkFieldNames.value === '' ? null : [linkFieldNames],
      }),
    );
    response = await JSON.parse(response);
    if (response.code === 400)
      Alert.alert('Something Went Wrong', response.message);
    else {
      PAGE_VARIABLES.postId = response.post.id;
      navigation.navigate('PostDetail');
    }
  }

  function getNameOfPost() {
    if (textFieldNames.name === '') return null;
    return (
      <View
        style={{
          margin: 5,
          flexDirection: 'row',
          alignItems: 'center',
          alignSelf: 'flex-start',
          width: '100%',
          borderColor: 'red',
        }}>
        {getName(textFieldNames.name)}
        <TextInput
          style={{
            height: 40,
            borderColor: 'gray',
            placeholder: 'Match Name',
            placeholderTextColor: 'gray',
            color: 'black',
            margin: 5,
            paddingBottom: 1,
          }}
          onChangeText={text =>
            setTextFieldNames({name: textFieldNames.name, value: text})
          }
          value={textFieldNames.value}
          placeholder="......."
        />
      </View>
    );
  }

  function getName(key) {
    return (
      <Text style={{color: 'black', fontSize: 20, fontWeight: 'bold'}}>
        {key}
      </Text>
    );
  }

  function getTextFieldNames() {
    if (textFieldNames.name === '') return null;
    return (
      <View
        style={{
          margin: 5,
          flexDirection: 'row',
          alignItems: 'center',
          alignSelf: 'flex-start',
        }}>
        {getName(textFieldNames.name)}
        <TextInput
          style={{
            height: 40,
            borderColor: 'gray',
            placeholder: 'Match Name',
            placeholderTextColor: 'gray',
            color: 'black',
            margin: 5,
            paddingBottom: 1,
          }}
          onChangeText={text =>
            setTextFieldNames({name: textFieldNames.name, value: text})
          }
          value={textFieldNames.value}
          placeholder="......."
        />
      </View>
    );
  }
  function getDate() {
    if (dateFieldNames.name === '') return null;
    return (
      <View
        style={{
          margin: 5,
          flexDirection: 'row',
          alignItems: 'center',
        }}>
        <Text style={{color: 'black', fontSize: 20, fontWeight: 'bold'}}>
          Date:
        </Text>
        <TextInput
          style={{
            height: 40,
            borderColor: 'gray',
            placeholder: 'Match Name',
            placeholderTextColor: 'gray',
            color: 'black',
            margin: 5,
            paddingBottom: 1,
          }}
          onChangeText={text =>
            setDateFieldNames({name: dateFieldNames.name, value: text})
          }
          value={dateFieldNames.value}
          placeholder="......."
        />
      </View>
    );
  }

  function getLink() {
    if (linkFieldNames.name === '') return null;
    return (
      <View
        style={{
          margin: 5,
          flexDirection: 'row',
          alignItems: 'center',
        }}>
        {getName(linkFieldNames.name)}
        <TextInput
          style={{
            height: 40,
            borderColor: 'gray',
            placeholder: 'Match Name',
            placeholderTextColor: 'gray',
            color: 'black',
            margin: 5,
            paddingBottom: 1,
          }}
          onChangeText={text =>
            setLinkFieldNames({name: linkFieldNames.name, value: text})
          }
          value={linkFieldNames.value}
          placeholder="......."
        />
      </View>
    );
  }
  function getNumbers() {
    if (numberFieldNames.name === '') return null;
    return (
      <View
        style={{
          margin: 5,
          flexDirection: 'row',
          alignItems: 'center',
        }}>
        {getName(numberFieldNames.name)}
        <TextInput
          style={{
            height: 40,
            borderColor: 'gray',
            placeholder: 'Match Name',
            placeholderTextColor: 'gray',
            color: 'black',
            margin: 5,
            paddingBottom: 1,
          }}
          onChangeText={text =>
            setNumberFieldNames({name: numberFieldNames.name, value: text})
          }
          value={numberFieldNames.value}
          placeholder="......."
        />
      </View>
    );
  }
  function getLocation() {
    if (locationFieldNames.name === '') return null;
    return (
      <View
        style={{
          margin: 5,
          flexDirection: 'row',
          alignItems: 'center',
        }}>
        {getName(locationFieldNames.name)}
        <TextInput
          style={{
            height: 40,
            borderColor: 'gray',
            placeholder: 'Match Name',
            placeholderTextColor: 'gray',
            color: 'black',
            margin: 5,
            paddingBottom: 1,
          }}
          onChangeText={text =>
            setLocationFieldNames({name: locationFieldNames.name, value: text})
          }
          value={locationFieldNames.value}
          placeholder="......."
        />
      </View>
    );
  }
  function myForum() {
    return (
      <View style={{alignContent: 'center'}}>
        <ScrollView showsVerticalScrollIndicator={false}>
          {getTextFieldNames()}
          {getDate()}
          {getLink()}
          {getLocation()}
          {getNumbers()}
        </ScrollView>
      </View>
    );
  }
  return (
    <View style={styles.container}>
      <View style={styles.header}>
        <TouchableOpacity onPress={() => navigation.goBack()}>
          <Icon name="close" size={34} style={{marginLeft: 5}}></Icon>
        </TouchableOpacity>
        <TouchableOpacity onPress={createPostHandler}>
          <View style={styles.nextButton}>
            <Text>Next</Text>
          </View>
        </TouchableOpacity>
      </View>
      <View style={styles.postTitleArea}>
        <View
          style={{
            width: 200,
            flexDirection: 'row',
            justifyContent: 'flex-start',
          }}>
          <Icon name="cube-sharp" size={20} style={{margin: 5}} />
          <Text style={{color: 'black', margin: 5}}>{communityName}</Text>
        </View>
        <Text>RULES</Text>
      </View>
      <View
        style={{
          minHeight: '50%',
          maxHeight: '50%',
          alignItems: 'flex-start',
        }}>
        {myForum()}
      </View>
    </View>
  );
}
const styles = {
  container: {flex: 1, height: '100%'},
  title: {},
  header: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  nextButton: {
    margin: 5,
    width: 70,
    height: 30,
    borderRadius: 8,
    backgroundColor: '#C5C6C6',
    alignItems: 'center',
    justifyContent: 'center',
  },
  postTitleArea: {
    justifyContent: 'space-between',
    marginHorizontal: 10,
    height: 30,
    borderBottomColor: 'grey',
    borderBottomWidth: 0.3,
    flexDirection: 'row',
  },
};
