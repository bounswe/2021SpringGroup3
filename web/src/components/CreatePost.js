import React from 'react';
import { useState, useEffect } from 'react';
import { useParams } from "react-router";
import { Row, Col, Form, Input, Typography, Button, Select, DatePicker, Card, Popover, Tag, Space } from 'antd';
import { LoadingOutlined } from '@ant-design/icons';
import { useSelector } from 'react-redux';
import { useDispatch } from 'react-redux';
import { useNavigate } from 'react-router';
import GetCommunities from './GetCommunities';
import GetPostTypes from './GetPostTypes';
import { CreatePost as CreatePostRequest } from '../utils/helper';
import { GetPostTypeDetail as GetPostTypeDetailRequest } from '../utils/helper';
import { SearchWikidata as SearchWikidataRequest } from '../utils/helper';

import MapPicker from 'react-google-map-picker'

const DefaultLocation = { lat: 41, lng: 29 };
const DefaultZoom = 10;

const { Text, Title } = Typography;
const { Option } = Select;

const CreatePost = (props) => {
  const loginState = useSelector((state) => state.login);
  const navigate = useNavigate();
  const dispatch = useDispatch();

  //react-map-picker
  const [defaultLocation, setDefaultLocation] = useState(DefaultLocation);
  const [location, setLocation] = useState(defaultLocation);
  const [zoom, setZoom] = useState(DefaultZoom);
  function handleChangeLocation(lat, lng) {
    setLocation({ lat: lat, lng: lng });
  }

  function handleChangeZoom(newZoom) {
    setZoom(newZoom);
  }

  function handleResetLocation() {
    setDefaultLocation({ ...DefaultLocation });
    setZoom(DefaultZoom);
  }

  const MapPickerElement =
    <div>
      <MapPicker defaultLocation={defaultLocation}
        zoom={zoom}
        mapTypeId="roadmap"
        style={{ height: '300px' }}
        onChangeLocation={handleChangeLocation}
        onChangeZoom={handleChangeZoom}
        apiKey={process.env.REACT_APP_GOOGLE_MAPS_API_KEY} />
    </div>

  const { communityId } = useParams();

  const postCardStyle = {
    width: '100%',
    backgroundColor: '#f8f8f8',
    borderRadius: '20px'
  }

  const buttonStyle = {
    backgroundColor: '#6f74dd',
    borderColor: '#6f74dd',
    color: '#ffffff',
    cursor: 'pointer',
    marginTop: '3px',
    marginBottom: '3px',
    fontWeight: 'bold'
  }

  const [textFieldsNames, setTextFieldsNames] = useState([]);
  const [numberFieldsNames, setNumberFieldsNames] = useState([]);
  const [linkFieldsNames, setLinkFieldsNames] = useState([]);
  const [dateFieldsNames, setDateFieldsNames] = useState([]);
  const [locationFieldsNames, setLocationFieldsNames] = useState([]);

  const [postTypeId, setPostTypeId] = useState('');
  const [postTypes, setPostTypes] = useState(<></>);
  const [textFields, setTextFields] = useState([]);
  const [numberFields, setNumberFields] = useState([]);
  const [linkFields, setLinkFields] = useState([]);
  const [dateFields, setDateFields] = useState([]);
  const [locationFields, setLocationFields] = useState([]);

  const onFinish = async (values) => {
    console.log("Success: ", communityId, postTypeId);
    let body = {
      communityId: communityId,
      postTypeId: postTypeId,
      textFields: [],
      numberFields: [],
      linkFields: [],
      dateFields: [],
      locationFields: []
    }

    console.log(values.tags)

    if (values.tags && values.tags.length > 0) {
      body.tags = values.tags.map(t => {
        let tag = JSON.parse(t)
        return {
          id: tag.id,
          name: tag.label
        }
      })
    }

    for (let key of Object.keys(values)) {
      if (textFieldsNames.includes(key)) {
        body.textFields.push({ name: key, value: values[key] })
      } else if (numberFieldsNames.includes(key)) {
        body.numberFields.push({ name: key, value: values[key] })
      } else if (linkFieldsNames.includes(key)) {
        body.linkFields.push({ name: key, value: values[key] })
      } else if (dateFieldsNames.includes(key)) {
        body.dateFields.push({ name: key, value: values[key]._d })
      } else if (locationFieldsNames.includes(key)) {
        body.locationFields.push({ name: key, value: { geo: { longitude: location.lng, latitude: location.lat }, description: "text" } })
      }
    }

    console.log(body)
    const result = await CreatePostRequest({ body: body, token: loginState.token }, dispatch);
    if (result && result.data) {
      navigate(`/communities/${result.data.community.id}/posts/${result.data.post.id}`)
    } else {
      navigate(`/communities/${communityId}`)
    }

  }

  const onFinishFailed = (error) => {
    console.log("Failed: ", error);
  }

  const resetFields = () => {
    setTextFields([]);
    setNumberFields([]);
    setLinkFields([]);
    setDateFields([]);
    setLocationFields([]);
    setTextFieldsNames([])
    setNumberFieldsNames([]);
    setLinkFieldsNames([]);
    setDateFieldsNames([]);
    setLocationFieldsNames([]);
  }

  useEffect(() => {
    if (communityId) {
      setPostTypes(
        <>
          <Col span={24}>
            <Text><b>Select Post Type to be Used</b></Text>
          </Col>
          <GetPostTypes id={communityId} isCreatePost={true} onSelectPostType={handlePostTypeChange} />
        </>
      )
    }
  }, [communityId])


  const handleCommunityChange = (selectedId) => {
    setPostTypes(<></>);
    setPostTypeId('');
    resetFields();
    navigate(`/createPost/community/${selectedId}`);
  }

  const handlePostTypeChange = (selectedId) => {
    resetFields();
    setPostTypeId(selectedId);
    GetPostTypeDetailRequest({ communityId: communityId, postTypeId: selectedId }, loginState.token, dispatch)
      .then(result => {
        setTextFieldsNames([...result.data.textFieldNames]);
        setNumberFieldsNames([...result.data.numberFieldNames]);
        setLinkFieldsNames([...result.data.linkFieldNames]);
        setDateFieldsNames([...result.data.dateFieldNames]);
        setLocationFieldsNames([...result.data.locationFieldNames]);

        setTextFields(result.data.textFieldNames.map(name => {
          return (
            <Row>
              <Col span={24}><Text strong>{name}</Text></Col>
              <Col span={24}>
                <Form.Item
                  name={name}
                  required
                  rules={[{ required: true, message: `Please enter ${name}` }]}>
                  <Input.TextArea type='text' placeholder={`Enter ${name}`} />
                </Form.Item></Col>
            </Row>
          )
        }))
        setNumberFields(result.data.numberFieldNames.map(name => {
          return (
            <Row key={name}>
              <Col span={24}><Text strong>{name}</Text></Col>
              <Col span={24}>
                <Form.Item
                  name={name}
                  required
                  rules={[{ required: true, message: `Please enter ${name}` }]}>
                  <Input type='number' placeholder={`Enter ${name}`} />
                </Form.Item></Col>
            </Row>
          )
        }))
        setLinkFields(result.data.linkFieldNames.map(name => {
          return (
            <Row key={name}>
              <Col span={24}><Text strong>{name}</Text></Col>
              <Col span={24}>
                <Form.Item
                  name={name}
                  required
                  rules={[{ required: true, message: `Please enter ${name}` }]}>
                  <Input type='text' placeholder={`Enter ${name}`} />
                </Form.Item>
              </Col>
            </Row>
          )
        }))
        setDateFields(result.data.dateFieldNames.map(name => {
          return (
            <Row key={name}>
              <Col span={24}><Text strong>{name}</Text></Col>
              <Col span={24}>
                <Form.Item
                  name={name}
                  required
                  rules={[{ required: true, message: `Please enter ${name}` }]}>
                  <DatePicker showTime />
                </Form.Item>
              </Col>
            </Row>
          )
        }))
        setLocationFields(result.data.locationFieldNames.map(name => {
          return (
            <Row key={name}>
              <Col span={24}><Text strong>{name}</Text></Col>
              <Col span={24}>
                <Form.Item
                  name={name}
                  required
                >
                  {MapPickerElement}
                </Form.Item>
              </Col>
            </Row>
          )
        }))
      })
  }

  const [tagDefinition, setTagDefinition] = useState({})
  const [searchedTags, setSearchedTags] = useState([])
  const [searchValue, setSearchValue] = useState([])

  const searchTags = async (value) => {
    setSearchValue(value)
  }

  useEffect(() => {
    SearchWikidataRequest({ tag: searchValue }, loginState.token, dispatch)
      .then(result => {
        if (result.data && result.data.length > 0) {
          console.log(result.data.map(t => t.description))
          setSearchedTags(result.data.map(t => { return { label: <Space direction='vertical' size={'0px'}><b>{t.label}</b>{t.description}</Space>, value: JSON.stringify(t) } }))
        } else {
          setSearchedTags([])
        }
      });
  }, [searchValue])

  const defineTag = async (tag) => {
    console.log(tag)
    const result = await SearchWikidataRequest({ tag: JSON.parse(tag.value).id }, loginState.token, dispatch)
    if (result.data && result.data.length > 0) {
      setTagDefinition(result.data[0])
    } else {
      setTagDefinition({ label: '', description: `No definition found for ${tag.label}.` })
    }
  }

  const cancelTag = () => {
    setTagDefinition({})
  }

  const tagRender = (tag) => {
    return (
      <Popover content={tagDefinition.description} title={tagDefinition.label}>
        <Tag
          color="#6f74dd"
          onMouseEnter={() => defineTag(tag)}
          onMouseLeave={() => cancelTag()}
          closable
          onClose={tag.onClose}
          style={{ cursor: 'pointer' }}>
          {JSON.parse(tag.value).label}
        </Tag>
      </Popover>
    )
  }

  return (
    <Card size="small" style={postCardStyle}>
      <Row>
        <Col span={10} offset={7} align="middle">
          <Title level={2} strong="true">Create a New Post</Title>
        </Col>

        <Col span={10} offset={7} style={{ marginTop: '20px' }} align="middle">
          <Col span={24}>
            <Text><b>Select Community to Post in</b></Text>
          </Col>
          <GetCommunities isCreatePost={true} onSelectCommunity={handleCommunityChange} />
        </Col>
        <Col span={10} offset={7} align="middle">
          {postTypes}
        </Col>

        <Col span={10} offset={7}>
          <Form name="basic" onFinish={onFinish} onFinishFailed={onFinishFailed}>
            {postTypeId ? <Col span={24} align="middle" style={{ marginBottom: '20px' }}>
              <Col span={24} align='middle'>
                <Text><b>Select Tags</b></Text>
              </Col>
              <Form.Item
                name='tags'>
                <Select
                  mode="multiple"
                  onSearch={(value) => { searchTags(value) }}
                  options={searchedTags}
                  tagRender={tagRender}
                  style={{ width: '100%' }}
                />
              </Form.Item>
            </Col> : <></>}
            {textFields}
            {numberFields}
            {linkFields}
            {dateFields}
            {locationFields}
            <Col span={24} align='middle'>
              {postTypeId ?
                <Form.Item>
                  <Button style={buttonStyle} type="primary" htmlType="submit" shape="round">
                    Submit Post
                  </Button>
                </Form.Item> : <></>
              }
            </Col>
          </Form>
        </Col>
      </Row>
    </Card>
  );
}

export default CreatePost;