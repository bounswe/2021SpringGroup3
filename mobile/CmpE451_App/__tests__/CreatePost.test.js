import React from 'react';
import CreatePost from '../src/screen/CreatePost';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(<CreatePost route={{params: {communityName: 'community'}}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
