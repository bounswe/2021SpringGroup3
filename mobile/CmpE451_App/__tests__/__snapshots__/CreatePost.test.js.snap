// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#eeeeee",
      "flex": 1,
      "height": "100%",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#00227b",
        "flexDirection": "row",
        "height": 50,
        "justifyContent": "space-between",
        "padding": 7,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "justifyItems": "space-between",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 6,
            "left": 6,
            "right": 6,
            "top": 6,
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "margin": 6,
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 22.5,
                "height": 45,
                "width": 45,
              },
              undefined,
              Object {
                "flex": 1,
              },
            ],
          ]
        }
      >
        <View>
          <Text
            accessibilityElementsHidden={true}
            allowFontScaling={false}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            selectable={false}
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 30,
                },
                Array [
                  Object {
                    "lineHeight": 30,
                    "transform": Array [
                      Object {
                        "scaleX": 1,
                      },
                    ],
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                ],
                Object {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            󰳝
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 5,
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 20,
            }
          }
        >
          Post to 
          community
        </Text>
      </View>
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 6,
            "left": 6,
            "right": 6,
            "top": 6,
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "margin": 6,
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 22.5,
                "height": 45,
                "width": 45,
              },
              undefined,
              Object {
                "flex": 1,
              },
            ],
          ]
        }
      >
        <View>
          <Text
            accessibilityElementsHidden={true}
            allowFontScaling={false}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            selectable={false}
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 30,
                },
                Array [
                  Object {
                    "lineHeight": 30,
                    "transform": Array [
                      Object {
                        "scaleX": 1,
                      },
                    ],
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                ],
                Object {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            󰳟
          </Text>
        </View>
      </View>
    </View>
  </View>
  <RCTScrollView
    showsVerticalScrollIndicator={false}
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "width": "100%",
              }
            }
          >
            <View />
            <View />
            <View />
            <View />
            <View />
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
